code-reviews/prepare-code-for-review.md

```diff
@@ -0,0 +1,58 @@
+# Prepare Code for Review
+
+## Program for the present, not the future
+
+At elementary, we believe in the YAGNI principle (You aren't gonna need it): Always implement when you _actually have_ a need - but never when you _think_ you are _gonna_ need it. Combined with continous refactoring, the YAGNI principle keeps the code base as simple as possible while helping us to avoid technical debt.
```

Comment:
> but never when you _think_ you are _gonna_ need it.

I think "never" is a bit strong - it depends on how justified your thinking is and over what timescale.  Maybe "never when you only feel it may be needed some time in the future".